test/solver/gmres_kernels.cpp

```diff
@@ -294,18 +339,27 @@ TEST_F(Gmres, GmresApplyOneRHSIsEquivalentToRef)
 
 TEST_F(Gmres, GmresApplyMultipleRHSIsEquivalentToRef)
 {
-    int m = 123;
-    int n = 5;
-    auto ref_solver = ref_gmres_factory->generate(mtx);
-    auto exec_solver = exec_gmres_factory->generate(d_mtx);
-    auto b = gen_mtx(m, n);
-    auto x = gen_mtx(m, n);
-    auto d_b = gko::clone(exec, b);
-    auto d_x = gko::clone(exec, x);
+    using gko::solver::gmres::orthog_method;
+    auto base_params = gko::clone(ref, ref_gmres_factory)->get_parameters();
 
-    ref_solver->apply(b, x);
-    exec_solver->apply(d_b, d_x);
-
-    GKO_ASSERT_MTX_NEAR(d_b, b, 0);
-    GKO_ASSERT_MTX_NEAR(d_x, x, r<value_type>::value * 1e3);
+    for (auto orthog :
+         {orthog_method::mgs, orthog_method::cgs, orthog_method::cgs2}) {
+        SCOPED_TRACE(orthog);
+        int m = 123;
+        int n = 5;
+        auto ref_solver =
+            base_params.with_orthog_method(orthog).on(ref)->generate(mtx);
+        auto exec_solver =
+            base_params.with_orthog_method(orthog).on(exec)->generate(d_mtx);
```

Comment:
Do you know if the default restart parameters are the ones being used ? I think it would be nice to test for problems that need higher restart parameters as well.

Reply:
Yeah, the default krylov_dim should be used from what I understand. The default is pretty high IMO (100) compared to the sizes that are used in the tests.

Reply:
The matrix used here is a dense random matrix of size (123x123) with values distributed from a normal distribution with mean -1.0 and stddev 1. I am not sure the edge cases of the different orthogonalization methods are being properly tested ?

Reply:
By "higher restart parameters" do you mean "needs to go all the way up to 100 (or higher if we set the dim higher)" or "needs to restart at least once" (which could be achieved with a small problem by dropping `krylov_dim`)?

Reply:
I think it needs to restart atleast once.

Reply:
I put a temporary convergence logger in the test and saw that it was actually not converging -- maxes out at 246 iterations (set as `max_iters`). I don't know if that's the intended behavior for this test, but the end result is that it should be restarting twice as-is. I output the matrix and rhs vectors that are generated and did some playing around in scipy. The matrix is fairly non-normal (Frobenius norm of A^TA - AA^T ~3452) and the GMRES convergence stagnates pretty severely after restart. Even with Krylov dim = 122 it stagnates around 10e-2 (in the scipy gmres); only going up to full size for the Krylov dim (123) gets that last "drop" (and it does, as it should).

Reply:
I see. Thanks for checking that. From the numbers you state, that should be sufficient then.